### ISSUE TITLE
ISSUE: Post DE counter is counting all mails in the defined inbox

### ISSUE THREAD

**Issue by eric1905** [moralmunky/Home-Assistant-Mail-And-Packages#832]
**Describe the bug**
Currently the sensor for Post DE is showing the number of all mails in the inbox, not just for the current day. This leads to a sensor value of 50, even if I received no letters today. On friday the counter was 48 and after that I received 2 "Briefankündigungen" and now it is 50. In the log it is searching for "SINCE 12-Jun-2023 SUBJECT" but the numbers in the next output seem to be total numbers as I have around 4200 mails in my inbox. So maybe the search is not working for me?
I am using GMX if this is needed.

**Environment (please complete the following information):**

 - Type of device: RPi4
 - Home Assistant 2023.6.1
 - Supervisor 2023.06.1
 - Operating System 10.2
 - Component version: 0.3.15

**Logs**
```
2023-06-12 14:11:16.166 DEBUG (SyncWorker_5) [custom_components.mail_and_packages.helpers] Unknown sensor type: post_de_delivered
2023-06-12 14:11:16.167 DEBUG (SyncWorker_5) [custom_components.mail_and_packages.helpers] Sensor: post_de_delivered Count: 0
2023-06-12 14:11:16.167 DEBUG (SyncWorker_5) [custom_components.mail_and_packages.helpers] Attempting to find mail from (['ankuendigung@brief.deutschepost.de']) with subject (Ein Brief kommt in Kürze bei Ihnen an)
2023-06-12 14:11:16.167 DEBUG (SyncWorker_5) [custom_components.mail_and_packages.helpers] DEBUG subject: Ein Brief kommt in Kürze bei Ihnen an
2023-06-12 14:11:16.167 DEBUG (SyncWorker_5) [custom_components.mail_and_packages.helpers] DEBUG imap_search: SINCE 12-Jun-2023 SUBJECT
2023-06-12 14:11:16.743 DEBUG (SyncWorker_5) [custom_components.mail_and_packages.helpers] email_search value: ('OK', [b'1451 1472 1479 1484 1487 1524 1553 1611 1676 1680 1690 1835 1966 2013 2163 2164 2192 2292 2324 2362 2404 2447 2490 2504 2640 2681 2797 2882 3048 3100 3160 3239 3300 3369 3382 3502 3530 3545 3577 3578 3731 3788 3903 3947 3948 4048 4125 4165 4184 4185'])
2023-06-12 14:11:16.744 DEBUG (SyncWorker_5) [custom_components.mail_and_packages.helpers] Search for (['ankuendigung@brief.deutschepost.de']) with subject (Ein Brief kommt in Kürze bei Ihnen an) results: b'1451 1472 1479 1484 1487 1524 1553 1611 1676 1680 1690 1835 1966 2013 2163 2164 2192 2292 2324 2362 2404 2447 2490 2504 2640 2681 2797 2882 3048 3100 3160 3239 3300 3369 3382 3502 3530 3545 3577 3578 3731 3788 3903 3947 3948 4048 4125 4165 4184 4185' count: 50
2023-06-12 14:11:16.744 DEBUG (SyncWorker_5) [custom_components.mail_and_packages.helpers] Sensor: post_de_delivering Count: 50
2023-06-12 14:11:16.745 DEBUG (SyncWorker_5) [custom_components.mail_and_packages.helpers] Sensor: post_de_packages Count: 50
```

**Comment by firstof9** [comment]
> 2023-06-12 14:11:16.167 DEBUG (SyncWorker_5) [custom_components.mail_and_packages.helpers] Attempting to find mail from (['[ankuendigung@brief.deutschepost.de](mailto:ankuendigung@brief.deutschepost.de)']) with subject (Ein Brief kommt in Kürze bei Ihnen an)
> 2023-06-12 14:11:16.167 DEBUG (SyncWorker_5) [custom_components.mail_and_packages.helpers] DEBUG subject: Ein Brief kommt in Kürze bei Ihnen an
> 2023-06-12 14:11:16.167 DEBUG (SyncWorker_5) [custom_components.mail_and_packages.helpers] DEBUG imap_search: SINCE 12-Jun-2023 SUBJECT
> 2023-06-12 14:11:16.743 DEBUG (SyncWorker_5) [custom_components.mail_and_packages.helpers] email_search value: ('OK', [b'1451 1472 1479 1484 1487 1524 1553 1611 1676 1680 1690 1835 1966 2013 2163 2164 2192 2292 2324 2362 2404 2447 2490 2504 2640 2681 2797 2882 3048 3100 3160 3239 3300 3369 3382 3502 3530 3545 3577 3578 3731 3788 3903 3947 3948 4048 4125 4165 4184 4185'])

There's no errors in between here?

**Comment by firstof9** [comment]
If you'd like to test you can make the changes specified in #833 locally and see if that helps.

**Comment by eric1905** [comment]
> If you'd like to test you can make the changes specified in #833 locally and see if that helps.

When I change the file helpers.py the component is not starting. I get the following errors after restarting home assistant:


2023-06-12 20:30:44.334 ERROR (SyncWorker_0) [custom_components.mail_and_packages.helpers] Error searching emails: command: SEARCH => too many errors
2023-06-12 20:30:44.341 ERROR (SyncWorker_0) [custom_components.mail_and_packages.helpers] Error searching emails: too many errors
2023-06-12 20:30:44.344 ERROR (SyncWorker_0) [custom_components.mail_and_packages.helpers] Error searching emails: too many errors
2023-06-12 20:30:44.347 ERROR (SyncWorker_0) [custom_components.mail_and_packages.helpers] Error searching emails: too many errors
2023-06-12 20:30:44.353 ERROR (SyncWorker_0) [custom_components.mail_and_packages.helpers] Error searching emails: too many errors
2023-06-12 20:30:44.357 ERROR (SyncWorker_0) [custom_components.mail_and_packages.helpers] Error searching emails: too many errors
2023-06-12 20:30:44.362 ERROR (SyncWorker_0) [custom_components.mail_and_packages.helpers] Error searching emails: socket error: TLS/SSL connection has been closed (EOF) (_ssl.c:2423)
2023-06-12 20:30:44.366 ERROR (SyncWorker_0) [custom_components.mail_and_packages.helpers] Error searching emails: socket error: [SSL: BAD_LENGTH] bad length (_ssl.c:2423)
2023-06-12 20:30:44.369 ERROR (SyncWorker_0) [custom_components.mail_and_packages.helpers] Error searching emails: socket error: [SSL: BAD_LENGTH] bad length (_ssl.c:2423)
2023-06-12 20:30:44.372 ERROR (SyncWorker_0) [custom_components.mail_and_packages.helpers] Error searching emails: socket error: [SSL: BAD_LENGTH] bad length (_ssl.c:2423)
2023-06-12 20:30:44.374 ERROR (SyncWorker_0) [custom_components.mail_and_packages.helpers] Error searching emails: socket error: TLS/SSL connection has been closed (EOF) (_ssl.c:2423)
2023-06-12 20:30:44.376 ERROR (SyncWorker_0) [custom_components.mail_and_packages.helpers] Error searching emails: socket error: [SSL: BAD_LENGTH] bad length (_ssl.c:2423)
2023-06-12 20:30:44.380 ERROR (SyncWorker_0) [custom_components.mail_and_packages.helpers] Error searching emails: socket error: [SSL: BAD_LENGTH] bad length (_ssl.c:2423)
2023-06-12 20:30:44.383 ERROR (SyncWorker_0) [custom_components.mail_and_packages.helpers] Error searching emails: socket error: [SSL: BAD_LENGTH] bad length (_ssl.c:2423)
2023-06-12 20:30:44.386 ERROR (SyncWorker_0) [custom_components.mail_and_packages.helpers] Error searching emails: socket error: [SSL: BAD_LENGTH] bad length (_ssl.c:2423)
2023-06-12 20:30:44.390 ERROR (SyncWorker_0) [custom_components.mail_and_packages.helpers] Error searching emails: socket error: TLS/SSL connection has been closed (EOF) (_ssl.c:2423)
2023-06-12 20:30:44.394 ERROR (MainThread) [custom_components.mail_and_packages] Problem updating sensors: 'count'
2023-06-12 20:30:44.399 ERROR (MainThread) [custom_components.mail_and_packages] Error fetching Mail and Packages (imap.gmx.net) data: 'count'
2023-06-12 20:30:44.401 ERROR (MainThread) [custom_components.mail_and_packages] Error updating sensor data: 'count'

2023-06-12 20:34:39.612 ERROR (SyncWorker_9) [custom_components.mail_and_packages.helpers] Error searching emails: command: SEARCH => too many errors
2023-06-12 20:34:39.616 ERROR (SyncWorker_9) [custom_components.mail_and_packages.helpers] Error searching emails: too many errors
2023-06-12 20:34:39.623 ERROR (SyncWorker_9) [custom_components.mail_and_packages.helpers] Error searching emails: too many errors
2023-06-12 20:34:39.630 ERROR (SyncWorker_9) [custom_components.mail_and_packages.helpers] Error searching emails: too many errors
2023-06-12 20:34:39.633 ERROR (SyncWorker_9) [custom_components.mail_and_packages.helpers] Error searching emails: too many errors
2023-06-12 20:34:39.645 ERROR (SyncWorker_9) [custom_components.mail_and_packages.helpers] Error searching emails: too many errors
2023-06-12 20:34:39.650 ERROR (SyncWorker_9) [custom_components.mail_and_packages.helpers] Error searching emails: too many errors
2023-06-12 20:34:39.654 ERROR (SyncWorker_9) [custom_components.mail_and_packages.helpers] Error searching emails: too many errors
2023-06-12 20:34:39.657 ERROR (SyncWorker_9) [custom_components.mail_and_packages.helpers] Error searching emails: too many errors
2023-06-12 20:34:39.661 ERROR (SyncWorker_9) [custom_components.mail_and_packages.helpers] Error searching emails: too many errors
2023-06-12 20:34:39.667 ERROR (SyncWorker_9) [custom_components.mail_and_packages.helpers] Error searching emails: too many errors
2023-06-12 20:34:39.673 ERROR (SyncWorker_9) [custom_components.mail_and_packages.helpers] Error searching emails: too many errors
2023-06-12 20:34:39.676 ERROR (SyncWorker_9) [custom_components.mail_and_packages.helpers] Error searching emails: too many errors
2023-06-12 20:34:39.679 ERROR (SyncWorker_9) [custom_components.mail_and_packages.helpers] Error searching emails: too many errors
2023-06-12 20:34:39.684 ERROR (SyncWorker_9) [custom_components.mail_and_packages.helpers] Error searching emails: too many errors
2023-06-12 20:34:39.688 ERROR (SyncWorker_9) [custom_components.mail_and_packages.helpers] Error searching emails: socket error: TLS/SSL connection has been closed (EOF) (_ssl.c:2423)
2023-06-12 20:34:39.692 ERROR (MainThread) [custom_components.mail_and_packages] Problem updating sensors: 'count'
2023-06-12 20:34:39.694 ERROR (MainThread) [custom_components.mail_and_packages] Error fetching Mail and Packages (imap.gmx.net) data: 'count'
2023-06-12 20:34:39.696 ERROR (MainThread) [custom_components.mail_and_packages] Error updating sensor data: 'count'

Do I need to compile the file before using it?

**Comment by eric1905** [comment]
2023-06-12 21:26:36.878 DEBUG (SyncWorker_8) [custom_components.mail_and_packages.helpers] DEBUG imap_search: (FROM "order-update@amazon.ca" SINCE 12-Jun-2023 SUBJECT)
2023-06-12 21:26:36.917 DEBUG (SyncWorker_8) [custom_components.mail_and_packages.helpers] Error searching emails with unicode characters: SEARCH command error: BAD [b'expected " " instead of ")"']
2023-06-12 21:26:36.917 DEBUG (SyncWorker_8) [custom_components.mail_and_packages.helpers] email_search value: ('BAD', 'SEARCH command error: BAD [b\'expected " " instead of ")"\']')

**Comment by firstof9** [comment]
> Do I need to compile the file before using it?

No

I've just pushed a new commit, please give that a try.

**Comment by firstof9** [comment]
Excellent, run it for a couple days (or however long it takes for you to get something in the post) and let me know. If everything's good, I'll merge the PR and release.

Thanks for testing!

**Comment by firstof9** [comment]
Fixed via #833
